src/app/_validation/PersonValidation.ts

```diff
@@ -1,12 +1,16 @@
+import { AbstractControl, ValidationErrors, ValidatorFn } from '@angular/forms';
 import { PersonInput } from '../../interfaces';
-export const isValidFirstName = (firstName: string): boolean => firstName?.length > 0;
+export const isValidFirstName = (): ValidatorFn => {
+    return (control: AbstractControl): ValidationErrors | null =>
+        control.value?.length > 0
+            ? null : { strLen: control.value.length };
```

Comment:
same actually here, you have minLength as well as a possible validators, 
let's reuse existing, 
you will find them here:
https://github.com/angular/angular/blob/257a30d7b394867be59dbdaf56cfe8e572c02fb8/packages/forms/src/validators.ts#L132

Reply:
also, just for the sake of knowledge, you can play with https://angular.io/api/forms/AsyncValidatorFn
make some simple promise just to see how Angular controls the state of form while validator is being resolved